staging/src/k8s.io/cloud-provider/fake/fake.go

```diff
@@ -362,11 +362,21 @@ func (f *Cloud) DeleteRoute(ctx context.Context, clusterName string, route *clou
 	f.Lock.Lock()
 	defer f.Lock.Unlock()
 	f.addCall("delete-route")
-	name := route.Name
-	if _, exists := f.RouteMap[name]; !exists {
-		f.Err = fmt.Errorf("no route found with name %q", name)
+	name := ""
+	for key, saved := range f.RouteMap {
```

Comment:
Are there any implications for other providers?

Reply:
No, the original implementation was wrong. The `route` in reality has no name, and should always compare using `target node` and `target cidr`.